packages/tools/kadena-cli/src/keys/commands/keysPlainGenerate.ts

```diff
@@ -1,62 +1,108 @@
-import { kadenaKeyPairsFromRandom } from '@kadena/hd-wallet';
+import { kadenaEncrypt, kadenaKeyPairsFromRandom } from '@kadena/hd-wallet';
+import { kadenaGenKeypair } from '@kadena/hd-wallet/chainweaver';
+import chalk from 'chalk';
+import type { Command } from 'commander';
+import { randomBytes } from 'crypto';
+import debug from 'debug';
 import { createCommand } from '../../utils/createCommand.js';
 import { globalOptions } from '../../utils/globalOptions.js';
+import { clearCLI } from '../../utils/helpers.js';
+import {
+  displayGeneratedPlainKeys,
+  printStoredPlainKeys,
+} from '../utils/keysDisplay.js';
+import { fromHexStr, toHexStr } from '../utils/keysHelpers.js';
+import * as storageService from '../utils/storage.js';
 
-import chalk from 'chalk';
-import debug from 'debug';
+interface IGeneratePlainKeysCommandConfig {
+  keyAlias: string;
+  keyAmount?: number;
+  keyPassword: string;
+  legacy?: boolean;
+}
 
-import { PLAINKEY_EXT } from '../../constants/config.js';
-import { clearCLI } from '../../utils/helpers.js';
+const defaultAmount: number = 1;
 
-import * as storageService from '../utils/storage.js';
+async function generateKeyPairs(
+  config: IGeneratePlainKeysCommandConfig,
+  amount: number,
+): Promise<storageService.IKeyPair[]> {
+  if (config.legacy === true) {
+    return generateLegacyKeyPairs(config, amount);
+  } else {
+    const randomKeyPairs = await kadenaKeyPairsFromRandom(amount);
+    return randomKeyPairs.map((keyPair) => {
+      const encryptedPrivateKey = kadenaEncrypt(
+        config.keyPassword,
+        fromHexStr(keyPair.secretKey),
+      );
+      return {
+        publicKey: keyPair.publicKey,
+        privateKey: encryptedPrivateKey,
```

Comment:
kda-tool uses `public` and `secret` should we do the same?

Reply:
i have taken this from hd-wallet, 
The question to support this like is done with kda-tool ( which is not used that much, i would prefer privateKey
@buckie  @emilypi  @sirlensalot  what do you think?

Reply:
generally cryptography libraries use secret, e.g. https://github.com/jedisct1/libsodium

Reply:
I would prefer `privateKey`. But if we go with `secretKey` we should annotate the argument to explain what secret key is referring to.

For context, I was working with some developers that were using our cryptography-utils library and they were confused on the difference between `private key` and `secret key`. They thought secret key might have been some kind of password/seed.

Reply:
I do not have hard evidence on this, but I feel like I've seen and used `secretKey` more than `privateKey` and so I would prefer `secretKey`. `tweetnacl` also uses `secretKey`: https://www.npmjs.com/package/tweetnacl#naclboxkeypair

I know @lsgunnlsgunn has been thinking about a glossary and how we name things. We should use the same wording everywhere. Lisa, what's your opinion on this?

Reply:
@LindaOrtega did the devs you worked with have any experience with cryptography? I wonder what eth web3.js calls it.

Reply:
@sirlensalot yes, they had some experience with cryptography and a lot of experience with Ethereum.

========

> I wonder what eth web3.js calls it.

I'm seeing `privatekey` used with Eth (see https://web3js.readthedocs.io/en/v1.10.0/web3-shh.html?highlight=keypair#addprivatekey)

This Node.js crypto library uses `privateKey` (see https://nodejs.org/api/crypto.html#signsignprivatekey-outputencoding).

But Solana uses `secretKey` (see https://solana-labs.github.io/solana-web3.js/classes/Keypair.html#fromSecretKey).

========
What is the difference between `secret key` and `private key`?
I've seen `secret key` and `private key` used interchangeably in the past, so I always assumed they were the same thing. I found [this](https://security.stackexchange.com/questions/90121/secret-public-and-private-keys) Stackoverflow answer on the difference between the two, but I'm unsure of its validity.

I don't have a preference between the two, but I +1 on having similar naming conventions across the org. 
I'm seeing `privatekey`, `privkey`, and `secretkey` used across kadena.js, pact, and chainweb-node repos (see [here](https://github.com/kadena-io/pact/blob/e5c922664fb4df53e3b8e2c48618d71b3ef612ad/src/Pact/Types/Crypto.hs#L171) for an example). 

Am I missing something? i.e. are we always using `secretkey` for external facing APIs?

Reply:
OK, I found some stuff, it looks like there is a lot of literature that uses secret for symmetric, and private for asymmetric, such as this internet glossary: https://datatracker.ietf.org/doc/html/rfc4949

Meanwhile all of the ED25519 djb literature exclusively uses "secret", the original paper literally does not have "private" in it.

I can only guess but I think the DJBs of this world are writing crypto libraries and want to audit secrets, whether asymmetric or no, for things like wiped memory etc. Meanwhile OpenSSL uses private key.

That (and Haskell crypto libs using secret) explains Pact, which was built around ED25519.

But it looks like the dispute is legitimate. Pact does have an "API" which is the yaml format and the output of `pact -g`, but otherwise private/secret keys never show up in an API for obvious reasons. 

I also note that this very codebase references existing Kadena code that uses "secret" (https://github.com/kadena-community/kadena.js/blob/a0c54b4bc9a20ff50c2fc589ec6ea3046cab27a8/packages/libs/hd-wallet/src/SLIP10/kadenaKeyPairsFromRandom.ts#L20) which itself uses node.js `crypto` which uses `privateKey` LOL

So both of the following are true:
1. Kadena more or less standardized on `secret`, from its direct influences (haskell, nacl, DJB); maybe the latter is Solana's influence too?
2. Non-DJB literature on asymmetric, plus most crypto projects (Bitcoin, Ethereum), use "private"

So I can now see either side and withdraw from this discussion :) However if you do end up using private, check in before you break our yaml format because we have numerous internal procedures that use that, which uses `secret`

Reply:
thanks for all the input @LindaOrtega @sirlensalot @alber70g @barthuijgen 
The team decided to go for secretKey to keep it consistent.